main.cpp

```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+extern "C" {
+#include "mmio.h"
+}
+
+void matrix_vector(int rows, int cols, double* A, double* v, double* out) {
+    for(int r=0; r<rows; r++) {
+        double rowSum = 0;
+        for(int c=0; c<cols; c++) {
+            rowSum += *(A+cols*r+c) * *(v+c);
```

Comment:
Rather than using pointer arithmetic in future assignments, we should use datastructures and provide appropriate call or index operators to use them. If pointer arithmetic is needed in C++ it often is indicitive of a problem.